### PR TITLE
Remove `astroquery/utils/testing_tools.py`

### LINKED ISSUE #1134: turn_on/off internet should use pytest-remotedata?
Should `astroquery.utils.testing_tools` use https://github.com/astropy/pytest-remotedata/blob/master/pytest_remotedata/disable_internet.py instead of implementing their own one-off enable/disable tools?

### PR DESCRIPTION
**Pull request by eerovaher**
The deleted file defined two pytest fixtures. Neither was used anywhere, and their functionality is provided by `pytest-remotedata`.

Closes #1134

### CODE REVIEW

**Review by keflavich — APPROVED** [review]
Thanks.  These were obsoleted by this functionality being incorporated somewhere into the astropy testing framework, though I don't remember where or when

**Comment by keflavich** [comment]
(my approval is contingent on tests passing)

**Comment by bsipocz** [comment]
Also a short changelog along the lines of "removal of obsolete testing tools" would be useful, added to the last changelog section.

File: astroquery/utils/testing_tools.py [hunk 1]
```diff
@@ -1,32 +0,0 @@
-# Licensed under a 3-clause BSD style license - see LICENSE.rst
-
-import socket
-
-import pytest
-
-# Import MockResponse to keep the API while it's temporarily factored out to
-# a separate file to avoid requiring pytest as a dependency in non-test code
-from .mocks import MockResponse
```
[the remark this answers is not in the document]
Comment (eerovaher) [reply astroquery/utils/testing_tools.py:9]: I ran the tests locally with
```shell
python setup.py test
```
and nothing failed. Running the tests with `tox` does lead to the same failures as were seen in CI.

[the remark this answers is not in the document]
Comment (bsipocz) [reply astroquery/utils/testing_tools.py:9]: I think some temporary files may have been left behind when you tested locally, but not when it was a fully independent build for tox. I expect that the testing would have failed after a `git clean`.

[the remark this answers is not in the document]
Comment (eerovaher) [reply astroquery/utils/testing_tools.py:9]: > I expect that the testing would have failed after a `git clean`.

Indeed they do.

**Review by bsipocz — APPROVED** [review]

**Comment by bsipocz** [comment]
Thanks @eerovaher!

**Pull request merged**